modules/git/submodule.go

```diff
@@ -51,8 +51,14 @@ func (sf *SubModuleFile) RefUrl() string {
 	i := strings.Index(url, "@")
 	j := strings.LastIndex(url, ":")
 	if i > -1 && j > -1 {
-		return "http://" + url[i+1:j] + "/" + url[j+1:]
+		// fix problem with reverse proxy works only with local server
+		if strings.Contains(appUrl,url[i+1:j]) {
```

Comment:
As a quick work around, why not just use `setting.AppUrl` here so you need to change anything in template file.